ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V102__add_initialization_indices.sql

```diff
@@ -0,0 +1,17 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+
+---------------------------------------------------------------------------------------------------
+-- V102: Add indices to speed up indexer initialization
+--
+-- At startup, the indexer deletes all entries with an offset beyond the stored ledger end.
+-- Such entries can be written when the indexer crashes right before updating the ledger end.
+-- This migration adds missing indices to speed up the deletion of such entries.
+---------------------------------------------------------------------------------------------------
+
+CREATE INDEX configuration_entries_ledger_offset_idx ON configuration_entries USING btree (ledger_offset);
```

Comment:
While adding this index, I realized that we don't prune configuration entries when we prune the participant. This seems to be a known shortcoming of the current pruning implementation.

Reply:
why is this needed? we have a primary key for the configuration_entries, which is the ledger_offset